src/main/java/org/terasology/web/resources/OnlinePlayersResource.java

```diff
@@ -52,11 +53,9 @@ public void onDisconnected(DisconnectedEvent event, EntityRef entityRef) {
     }
 
     @Override
-    public List<String> read(Client requestingClient) {
-        List<String> result = new ArrayList<>();
-        for (Client client: networkSystem.getPlayers()) {
-            result.add(client.getName());
-        }
-        return result;
+    public List<OnlinePlayerMetadata> read(Client requestingClient) {
+        return StreamSupport.stream(networkSystem.getPlayers().spliterator(), true)
```

Comment:
Can you not just call `getPlayers().stream()` here instead? Sorry, I don't have the IDE here, atm.